src/tasks/DockerBuildTaskProvider.ts

```diff
@@ -25,59 +25,38 @@ export interface DockerBuildTask extends Task {
     definition: DockerBuildTaskDefinition;
 }
 
-export class DockerBuildTaskProvider implements TaskProvider {
-    constructor(private readonly helpers: { [key in DockerPlatform]: TaskHelper }) { }
-
-    public provideTasks(token?: CancellationToken): ProviderResult<Task[]> {
-        return []; // Intentionally empty, so that resolveTask gets used
-    }
-
-    public resolveTask(task: DockerBuildTask, token?: CancellationToken): ProviderResult<Task> {
-        return callWithTelemetryAndErrorHandling(
-            'docker-build-resolve',
-            async (actionContext: IActionContext) => {
-                const taskPlatform = getPlatform(task.definition);
-                actionContext.telemetry.properties.platform = taskPlatform;
-
-                return await this.resolveTaskInternal(
-                    {
-                        folder: task.scope as WorkspaceFolder,
-                        platform: taskPlatform,
-                        actionContext: actionContext,
-                        cancellationToken: token,
-                    },
-                    task
-                );
-            }
-        );
-    }
+export class DockerBuildTaskProvider extends DockerTaskProviderBase {
+    constructor(helpers: { [key in DockerPlatform]: TaskHelper }) { super('docker-build', helpers) }
 
     // TODO: Skip if image is freshly built
-    private async resolveTaskInternal(context: DockerBuildTaskContext, task: DockerBuildTask): Promise<Task> {
+    protected async executeTaskInternal(context: DockerBuildTaskContext, task: DockerBuildTask): Promise<void> {
         const definition = cloneObject(task.definition);
         definition.dockerBuild = definition.dockerBuild || {};
 
-        if (!context.folder) {
-            throw new Error(`Unable to determine task scope to execute docker-build task '${task.name}'.`);
-        }
-
         const helper = this.getHelper(context.platform);
 
-        if (helper) {
-            definition.dockerBuild = await helper.resolveDockerBuildOptions(context, definition);
+        if (helper && helper.preBuild) {
+            await helper.preBuild(context, definition);
+            throwIfCancellationRequested(context);
         }
 
-        await this.validateResolvedDefinition(context, definition.dockerBuild);
+        if (helper) {
+            definition.dockerBuild = await helper.getDockerBuildOptions(context, definition);
+            await this.validateResolvedDefinition(context, definition.dockerBuild);

```

Comment:
Is there benefit in validating the definition even when there's no helper?  I.e. would we catch any mis-configuration by not specific to a platform?

Reply:
I  meant to keep validation either way so this was an accident. I'll make this change.